backend/extensionController/BucketFsApi.go

```diff
@@ -0,0 +1,93 @@
+package extensionController
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+)
+
+// BucketFsAPI allows access to BucketFS. Currently, it's implemented by running UDFs via SQL that read the data. In the future that implementation might be replaced by direct access.
+type BucketFsAPI interface {
+	// ListBuckets returns a list of public buckets
+	ListBuckets() ([]string, error)
+	// ListFiles lists the files in a given bucket
+	ListFiles(bucket string) ([]BfsFile, error)
+}
+
+// CreateBucketFsAPI create an instance of BucketFsAPI
+func CreateBucketFsAPI(db *sql.DB) BucketFsAPI {
+	bucketFsAPI := bucketFsAPIImpl{db: db}
+	return &bucketFsAPI
+}
+
+type bucketFsAPIImpl struct {
+	db *sql.DB
+}
+
+func (bfsApi bucketFsAPIImpl) ListBuckets() ([]string, error) {
+	files, err := bfsApi.listDirInUDF("/buckets/bfsdefault/")
+	if err != nil {
+		return nil, err
+	}
+	names := make([]string, 0, len(files))
+	for _, file := range files {
+		names = append(names, file.Name)
+	}
+	return names, nil
+}
+
+func (bfsApi bucketFsAPIImpl) ListFiles(bucket string) ([]BfsFile, error) {
+	if strings.Contains(bucket, "/") {
+		return nil, fmt.Errorf("invalid bucket name. Bucket name must not contain slashes")
+	}
+	return bfsApi.listDirInUDF("/buckets/bfsdefault/" + bucket)
+}
+
+// BfsFile represents a file in BucketFS
+type BfsFile struct {
+	Name string
+	Size int
+}
+
+func (bfsApi bucketFsAPIImpl) listDirInUDF(directory string) ([]BfsFile, error) {
+	transaction, err := bfsApi.db.Begin()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a transaction. Cause: %w", err)
+	}
+	_, err = transaction.Exec("CREATE SCHEMA INTERNAL")
```

Comment:
Do we need to use a unique schema name?
Maybe create a ticket to do this later.